src/main/java/com/dataloom/hazelcast/ListenableHazelcastFuture.java

```diff
@@ -0,0 +1,57 @@
+package com.dataloom.hazelcast;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.hazelcast.core.ExecutionCallback;
+import com.hazelcast.core.ICompletableFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class ListenableHazelcastFuture<T> implements ListenableFuture<T> {
+    private static final Logger logger = LoggerFactory.getLogger( ListenableHazelcastFuture.class );
+    private final ICompletableFuture<T> f;
+
+    public ListenableHazelcastFuture( ICompletableFuture<T> f ) {
+        this.f = f;
+    }
+
+    @Override
+    public void addListener( Runnable listener, Executor executor ) {
+        f.andThen( new ExecutionCallback<T>() {
+            @Override public void onResponse( T response ) {
+                listener.run();
+            }
+
+            @Override public void onFailure( Throwable t ) {
+                logger.error( "Unable to retrieve Ace.", t );
```

Comment:
we probably should change "Ace" to something else, since `AceFuture` is not the only thing using this.